eng/pipelines/templates/steps/vmr-prepare.yml

```diff
@@ -28,11 +33,12 @@ steps:
     path: vmr
     clean: true
 
-- script: |
-    git checkout --track origin/${{ parameters.vmrBranch }}
```

Comment:
the problem with `Build.SourceBranchName` is that it has [bad behavior](url) when the branch name has a slash:

> Git repo branch, pull request, or tag: The last path segment in the ref. For example, in refs/heads/main this value is main. **In refs/heads/feature/tools this value is tools.**

There are numerous issues about this on various AzDO issue trackers but it's not going to get fixed.
We actually hit this in this repo a while ago: https://github.com/dotnet/installer/pull/15377

Reply:
Ahh yeah, you're right. You do have 'replace' available https://learn.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops#replace

Reply:
that unfortunately also doesn't work here since the parameter is initialized with the raw string `$(Build.SourceBranch)`, not the actual value so the `replace` doesn't work.

Reply:
Hmm...does it work if you set a variable based on the parameter, then use the following format to do the replace?

value: ${{ replace(replace(variables['vmrBranchNameVar'], 'refs/heads/', ''), 'refs/pull/', '') }}

But yeah a bit awkward.